### DISCUSSION TITLE
[setValue] inconsistent behavior with `setValue` and errors

### DISCUSSION THREAD

**Discussion by kuserich** [react-hook-form/react-hook-form#9824] [Q&A] [answered]
### Version Number

7.42.1

### Codesandbox/Expo snack

https://codesandbox.io/s/unruffled-fog-6wof3u

### Steps to reproduce

**Simple case works fine**
1. Go to [CodeSandbox](https://codesandbox.io/s/unruffled-fog-6wof3u) ([app](https://6wof3u.csb.app/))
2. Drag and drop any XML file on the yellow div with the `drop file here` text
3. Submit the form (there are no errors and you will see a `Form was submitted` message)

**Trying a wrong file first breaks the form**
1. Refresh the page to get a clean setup
2. Drag and drop any **non XML** file on the yellow div with the `drop file here` text (you will see the validation error)
3. Click the submit button
4. Drag and drop an XML file on the yellow div
5. Click the submit button

The form is now broken. The drag and drop feature won't work and the form data seems to be empty upon submission. Selecting a file using the browser file picker (i.e. using the `file` input) seems to fix the issue.

**Not submitting seems to work**
1. Refresh the page to get a clean setup
2. Drag and drop any **non XML** file on the yellow div with the `drop file here` text (you will see the validation error)
3. Do not submit but drag and drop any XML file on the yellow div
4. Submit the form

There is an additional special case when the input field is validated with `required`, the drag and drop field never works (uncomment the line `required` in CodeSandbox:
```JS
<input
    type="file"
    {...(register &&
        register("file", {
            required: "This field is required",
            validate: {
                isXml: (file) =>
                    (file[0] && file[0].type === "text/xml") || "File must be XML"
                }
            }))}
/>
```

### Expected behaviour

- `setValue` should set the value of the `file` input field correctly after a failed submission
- `required` should not fail after a file was set using `setValue`
- form data should not be empty after a failed submission

### What browsers are you seeing the problem on?

Chrome

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by bluebill1049** [comment, marked as answer]
`setValue` is not supported with file input, that's a browser limitation with file object.
